### PR TITLE
WIP: Upgrade to version 1.18.0

### PR DESCRIPTION
**Pull request by quaresmajose**
This pull request It was also submitted to OE-Core
https://patchwork.openembedded.org/series/26619

I have marked this as WIP because it has not yet been accepted in OE-Core,
It can be useful for someone and can also be tested by a larger group of users

### CODE REVIEW

**Comment by otavio** [comment]
Awesome @quaresmajose; I marked @dv1 and @fbertux to review it as well.

File: recipes-multimedia/gstreamer/gstreamer1.0_1.18.0.bb [hunk 1]
```diff
@@ -6,41 +6,38 @@ BUGTRACKER = "https://bugzilla.gnome.org/enter_bug.cgi?product=Gstreamer"
...
+           "
+SRC_URI[md5sum] = "25d683630ff8b82b2c878aa3edd3c491"
+SRC_URI[sha256sum] = "0ff09245b06c0aeb5d9a156edcab088a7e8213a0bf9c84a1ff0318f9c00c7805"
 
 PACKAGECONFIG ??= "${@bb.utils.contains('PTEST_ENABLED', '1', 'tests', '', d)} \
                    check \
                    debug \
-                   tools"
+                   tools \
+                   capabilities"
```
Comment (otavio) [inline recipes-multimedia/gstreamer/gstreamer1.0_1.18.0.bb]: The capabilities addition should be added as another patch as it is changing the current build settings.

File: recipes-multimedia/gstreamer/gstreamer1.0-plugins-good_1.18.0.bb [hunk 2]
```diff
@@ -1,19 +1,16 @@
...
-SRC_URI[md5sum] = "bd025f8f14974f94b75ac69a9d1b9c93"
-SRC_URI[sha256sum] = "40bb3bafda25c0b739c8fc36e48380fccf61c4d3f83747e97ac3f9b0171b1319"
+SRC_URI[md5sum] = "c437bf42bf830bc0f482c7d088fb37a5"
+SRC_URI[sha256sum] = "6329bc377a3afea5cf1ab329a91cc4a906f075781780d1edc59e8bea8c1a8bac"
 
 S = "${WORKDIR}/gst-plugins-good-${PV}"
 
 LICENSE = "GPLv2+ & LGPLv2.1+"
 LIC_FILES_CHKSUM = "file://COPYING;md5=a6f89e2100d9b6cdffcea4f398e37343 \
-                    file://common/coverage/coverage-report.pl;beginline=2;endline=17;md5=a4e1830fce078028c8f0974161272607 \
```
Comment (otavio) [inline recipes-multimedia/gstreamer/gstreamer1.0-plugins-good_1.18.0.bb:16]: This removal ought to be noted on the commit log.

File: recipes-multimedia/gstreamer/gstreamer1.0-plugins-good_1.18.0.bb [hunk 3]
```diff
@@ -23,13 +20,15 @@ RPROVIDES_${PN}-soup += "${PN}-souphttpsrc"
 PACKAGECONFIG ??= " \
     ${GSTREAMER_ORC} \
     ${@bb.utils.filter('DISTRO_FEATURES', 'pulseaudio x11', d)} \
+    ${@bb.utils.contains('TUNE_FEATURES', 'm64', 'asm', '', d)} \
```
Comment (otavio) [inline recipes-multimedia/gstreamer/gstreamer1.0-plugins-good_1.18.0.bb:23]: Please, if this is required for the new version, please add to the commit. Otherwise, if it is an improvement, it ought to be done as another commit.

File: recipes-multimedia/gstreamer/gstreamer1.0-plugins-good_1.18.0.bb [hunk 4]
```diff
@@ -62,6 +62,7 @@ EXTRA_OEMESON += " \
     -Doss4=disabled \
     -Dosxaudio=disabled \
     -Dosxvideo=disabled \
+    -Drpicamsrc=disabled \
```
Comment (otavio) [inline recipes-multimedia/gstreamer/gstreamer1.0-plugins-good_1.18.0.bb:67]: New commit, please.

File: recipes-multimedia/gstreamer/gstreamer1.0-plugins-bad_1.18.0.bb [hunk 5]
```diff
@@ -93,18 +91,22 @@ PACKAGECONFIG[zbar]            = "-Dzbar=enabled,-Dzbar=disabled,zbar"
...
 
 EXTRA_OEMESON += " \
+    -Ddoc=disabled \
     -Ddecklink=enabled \
     -Ddvb=enabled \
     -Dfbdev=enabled \
     -Dipcpipeline=enabled \
     -Dnetsim=enabled \
     -Dshm=enabled \
+    -Dv4l2codecs=enabled \
```
Comment (otavio) [inline recipes-multimedia/gstreamer/gstreamer1.0-plugins-bad_1.18.0.bb]: Those ought to be new PACKAGECONFIG options, no? also, in a new commit.

File: recipes-multimedia/gstreamer/gstreamer1.0-vaapi_1.18.0.bb [hunk 6]
```diff
@@ -8,28 +8,24 @@ REALPN = "gstreamer-vaapi"
...
 
-inherit meson pkgconfig gtk-doc features_check upstream-version-is-even
+inherit meson pkgconfig features_check upstream-version-is-even
 
 REQUIRED_DISTRO_FEATURES ?= "opengl"
 
 EXTRA_OEMESON += " \
+    -Ddoc=disabled \
     -Dexamples=disabled \
+    -Dtests=disabled \
```
Comment (otavio) [inline recipes-multimedia/gstreamer/gstreamer1.0-vaapi_1.18.0.bb]: This seems to break the tests package, below.

File: recipes-multimedia/gstreamer/gst-examples_1.18.0.bb [hunk 7]
```diff
@@ -2,14 +2,14 @@ SUMMARY = "GStreamer examples (including gtk-play, gst-play)"
 LICENSE = "LGPL-2.0+"
 LIC_FILES_CHKSUM = "file://playback/player/gtk/gtk-play.c;beginline=1;endline=20;md5=f8c72dae3d36823ec716a9ebcae593b9"
 
-DEPENDS = "glib-2.0 gstreamer1.0 gstreamer1.0-plugins-base gstreamer1.0-plugins-bad gtk+3 glib-2.0-native"
+DEPENDS = "glib-2.0 gstreamer1.0 gstreamer1.0-plugins-base gstreamer1.0-plugins-bad gtk+3 libsoup-2.4 json-glib glib-2.0-native"
```
Comment (otavio) [inline recipes-multimedia/gstreamer/gst-examples_1.18.0.bb:5]: Please mention new dependencies on the commit log.

**Review by otavio — CHANGES_REQUESTED** [review]

File: recipes-multimedia/gstreamer/gstreamer1.0_1.18.0.bb (hunk 1 above)
Comment (quaresmajose) [reply recipes-multimedia/gstreamer/gstreamer1.0_1.18.0.bb]: I have mention this change on the commit message
```
    * capfix.patch
      Removed as the new PACKAGECONFIG capabilities can be used instead to achieve the same goals.
      Fix reproducibility issue around libcap.
```
But after review this again it has a different behavior in this patch set

On the 1.16 it uses 'setuid-root'` if the 'setcap' is not provided in PACKAGECONFIG
I will need to do the same behavior and remove the new 'capabilities' from default PACKAGECONFIG

File: recipes-multimedia/gstreamer/gstreamer1.0-plugins-good_1.18.0.bb (hunk 2 above)
Comment (quaresmajose) [reply recipes-multimedia/gstreamer/gstreamer1.0-plugins-good_1.18.0.bb:16]: I will fix this

File: recipes-multimedia/gstreamer/gstreamer1.0-plugins-good_1.18.0.bb (hunk 3 above)
Comment (quaresmajose) [reply recipes-multimedia/gstreamer/gstreamer1.0-plugins-good_1.18.0.bb:23]: It is a new meson option on this version.
I will mention this on the commit message

File: recipes-multimedia/gstreamer/gstreamer1.0-plugins-good_1.18.0.bb (hunk 4 above)
Comment (quaresmajose) [reply recipes-multimedia/gstreamer/gstreamer1.0-plugins-good_1.18.0.bb:67]: I will do it

File: recipes-multimedia/gstreamer/gstreamer1.0-plugins-bad_1.18.0.bb (hunk 5 above)
Comment (quaresmajose) [reply recipes-multimedia/gstreamer/gstreamer1.0-plugins-bad_1.18.0.bb]: <del>This don't have any dependencies on external packages.

I will add a `PACKAGECONFIG` for this because it depends on `libgudev`

File: recipes-multimedia/gstreamer/gstreamer1.0-vaapi_1.18.0.bb (hunk 6 above)
Comment (quaresmajose) [reply recipes-multimedia/gstreamer/gstreamer1.0-vaapi_1.18.0.bb]: probably yes, I will enable it

File: recipes-multimedia/gstreamer/gst-examples_1.18.0.bb (hunk 7 above)
Comment (quaresmajose) [reply recipes-multimedia/gstreamer/gst-examples_1.18.0.bb:5]: I will do it

File: recipes-multimedia/gstreamer/gstreamer1.0_1.18.0.bb (hunk 1 above)
Comment (quaresmajose) [reply recipes-multimedia/gstreamer/gstreamer1.0_1.18.0.bb]: I can remove the PACKAGECONFIG `capabilities` and use the  `setcap`  to do the same.
With this changes I only drop the patch `capfix.patch` and adjusts the `setcap`.

```patch
-PACKAGECONFIG[setcap] = "-Dsetcap=enabled,-Dsetcap=disabled,libcap libcap-native"
+PACKAGECONFIG[setcap] = ",,libcap libcap-native"
```

Comment (quaresmajose) [reply recipes-multimedia/gstreamer/gstreamer1.0_1.18.0.bb]: I also remove `libcap` from `DEPENDS` because it is only needed to set the capabilities.
Now it as the same behavior of the version 1.16.

**Review by otavio — APPROVED** [review]
